food-truck.eve.md

```diff
@@ -2,91 +2,105 @@
+      wouldn't it be better to just not draw?
     - If location is given, put a pin
       - Clicking the map opens up default map program
   - Draw the menu title
-    - Draw the shopping cart icon in corner
-      - Default is showing no number next to icon
-      - As items are added, count the items and display the count next to icon
-      - Clicking the icon takes you to the checkout page
-  - Draw menu items
-    - Draw photo
-    - Draw item name
-    - Swiping right adds one to the user cart
-    - Swiping left removes one from the user cart
-  - Draw a user queue banner at the very top if an order is pending for them
 
-### Sample Menu
+Draw the menu items with image, name, cost, addition button
+```
+search
+   [#app page:"homepage"]
+   menu-item = [#menu name image cost]
+bind @browser
+   [#div menu-item #container children:(
+      [#div text:"{{name}} - ${{cost}}"]
+      [#div name style:[width:"auto" height:"auto" max-width:"200px" display:"block" content:"url({{image}})"]]
+      [#div menu-item #add text:"add to order!" style:[border:"2px solid black" width:80]]
+     )]
 ```
-commit 
-   [#menu name:"Bacon Swiss Burger"
-    image:"https://pixabay.com/en/burger-food-meat-tasty-1428948/"
-    description:"A half pound burger made from Niman Ranch beef with melted Swiss, thick cut bacon, and housemade aioli and ketchup."
-    cost:12]
-
-   [#menu name:"Veggie Burger"
-    image:"https://pixabay.com/en/eat-burger-onion-avocado-veggi-1352880/"
-    description:"Our totally vegan black bean and portabella mushroom burger, topped with avocado, grilled onion, vegan cheese, and soy mayo, all on a gluten-free bun"
-    cost:12
-    #gluten-free
-    #vegetarian]
-
-   [#menu name:"Chicken Salad Sandwich"
-    image:"https://pixabay.com/en/bread-breakfast-brown-business-1239276/"
-    description:"Grandma’s chicken salad recipe with grapes and walnuts, served on wholesome 12 grain bread"
-    cost:10]
 
-   [#menu
-    name:"Charbroiled Chicken Wings"
-    image:"https://pixabay.com/en/barbecue-bbq-charcoal-chicken-88340/"
-    description:"Brined for 24 hours and coated with our secret spice rub, then grilled until then skin is crispy and the meat is juicy"
-    cost:10
-    #gluten-free
-    #spicy]
+Draw the remove-from-cart button for each menu item if this item is in the cart
+```
+search
+   [#app page:"homepage" order]
+   order-item = [#order-item order menu-item count]
+   not (count = 0)
+search @browser
+   d = [#div menu-item #container]
+bind @browser
+   d.children += [#div order-item #remove text:"remove from order" style:[border:"2px solid black" width:80]]
+```
 
+Add an item to the order after a click. in the original design this was a swipe right
+```
+search @browser @event @session
+   [#app page:"homepage" order]
+   [#click element:[#add menu-item]]
+   count = if [#order-item order menu-item count:c] then c + 1 else 1
+commit
+   order-item = [#order-item order menu-item]
+   order-item.count := count
```

Comment:
any reason this is separate instead of part of order item? to keep it out of the identity we can use the `|`